src/devices/video/nt7534.cpp

```diff
@@ -0,0 +1,355 @@
+
+nt7534_device::nt7534_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
+	: nt7534_device(mconfig, NT7534, tag, owner, clock)
+{
+}
+
+nt7534_device::nt7534_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock)
+	: device_t(mconfig, type, tag, owner, clock)
+{
+}
+
+//-------------------------------------------------
+//  device_start - device-specific startup
+//-------------------------------------------------
+
+void nt7534_device::device_start()
+{
+	m_busy_timer = timer_alloc(TIMER_BUSY);
+
+	// state saving
+	save_item(NAME(m_busy_flag));
+	save_item(NAME(m_page));
+	save_item(NAME(m_column));
+	save_item(NAME(m_dr));
+	save_item(NAME(m_ir));
+	save_item(NAME(m_display_on));
+	save_item(NAME(m_display_start_line));
+	save_item(NAME(m_direction));
+	save_item(NAME(m_data_len));
+	save_item(NAME(m_ddram));
+	save_item(NAME(m_nibble));
+	save_item(NAME(m_rs_state));
+	save_item(NAME(m_rw_state));
+	save_item(NAME(m_backup_column));
+	save_item(NAME(m_read_modify_write));
+	save_item(NAME(m_adc));
+	save_item(NAME(m_reverse));
+	save_item(NAME(m_entire_display_on));
+}
+
+//-------------------------------------------------
+//  device_reset - device-specific reset
+//-------------------------------------------------
+
+void nt7534_device::device_reset()
+{
+	memset(m_ddram, 0x00, sizeof(m_ddram));
+
+	m_page       = 0;
+	m_column     = 0;
+	m_dr         = 0;
+	m_ir         = 0;
+	m_display_on = false;
+	m_entire_display_on = false;
+	m_direction  = 1;
+	m_adc        = true;
+	m_read_modify_write = false;
+	m_reverse  = false;
+	m_data_len   = 8;
+	m_nibble     = false;
+	m_rs_state   = 0;
+	m_rw_state   = 0;
+
+	set_busy_flag(1520);
+}
+
+
+//-------------------------------------------------
+//  device_timer - handler timer events
+//-------------------------------------------------
+
+void nt7534_device::device_timer(emu_timer &timer, device_timer_id id, int param, void *ptr)
+{
+	switch (id)
+	{
+		case TIMER_BUSY:
+			m_busy_flag = false;
+			break;
+	}
+}
+
+
+//**************************************************************************
+//  HELPERS
+//**************************************************************************
+
+void nt7534_device::set_busy_flag(uint16_t usec)
+{
+	m_busy_flag = true;
+	m_busy_timer->adjust( attotime::from_usec( usec ) );
+}
+
+void nt7534_device::update_nibble(int rs, int rw)
+{
+	if (m_rs_state != rs || m_rw_state != rw)
+	{
+		m_rs_state = rs;
+		m_rw_state = rw;
+		m_nibble = false;
+	}
+
+	m_nibble = !m_nibble;
+}
+
+//**************************************************************************
+//  device interface
+//**************************************************************************
+
+uint32_t nt7534_device::screen_update(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect)
+{
+	bitmap.fill(0, cliprect);
+
+	uint16_t px;
+	uint8_t py;
+	for (uint16_t x = 0; x < 132; x++){
```

Comment:
Could you please be consistent with K&R vs Allman style within a file?  This function is using K&R style (opening/closing brace on same line as control statement) while `nt7534_device::update_nibble` and `nt7534_device::device_timer` are using Allman style (opening/closing brace on separate line).